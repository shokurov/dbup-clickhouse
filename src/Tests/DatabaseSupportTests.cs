using DbUp.Builder;
using DbUp.Tests.Common;

namespace DbUp.ClickHouse.Tests;

public class DatabaseSupportTests : DatabaseSupportTestsBase
{
    public DatabaseSupportTests() : base()
    {
    }

    protected override UpgradeEngineBuilder DeployTo(SupportedDatabases to)
        => to.ClickHouseDatabase("");

    protected override UpgradeEngineBuilder AddCustomNamedJournalToBuilder(UpgradeEngineBuilder builder, string schema, string tableName)
        => builder.JournalTo(
            (connectionManagerFactory, logFactory)
                => new ClickHouseJournal(connectionManagerFactory, logFactory, schema, tableName)
        );
}
