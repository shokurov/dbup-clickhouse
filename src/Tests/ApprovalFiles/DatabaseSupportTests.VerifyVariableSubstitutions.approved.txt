DB Operation: Open connection
Info:         Beginning database upgrade
Info:         Checking whether journal table exists
DB Operation: Execute scalar command: EXISTS TABLE schemaversions
DB Operation: Dispose command
Info:         Journal table does not exist
Info:         Executing Database Server script 'Script0001.sql'
Info:         Checking whether journal table exists
DB Operation: Execute scalar command: EXISTS TABLE schemaversions
DB Operation: Dispose command
Info:         Creating the `schemaversions` table
DB Operation: Execute non query command: CREATE TABLE `schemaversions`
(
    ScriptName String,
    Applied DateTime
)
ENGINE = MergeTree()
ORDER BY (ScriptName)
DB Operation: Dispose command
Info:         The `schemaversions` table has been created
DB Operation: Execute non query command: print SubstitutedValue
DB Operation: Dispose command
DB Operation: Create parameter
Info:         DB Operation: Add parameter to command: scriptName=Script0001.sql
DB Operation: Create parameter
Info:         DB Operation: Add parameter to command: applied=<date>
DB Operation: Execute non query command: INSERT INTO `schemaversions` (ScriptName, Applied) VALUES (@scriptName, @applied)
DB Operation: Dispose command
Info:         Upgrade successful
DB Operation: Dispose connection
